core/src/main/java/org/mule/runtime/core/internal/processor/strategy/StreamPerThreadSink.java

```diff
@@ -115,8 +175,10 @@ public BackPressureReason emit(CoreEvent event) {
   @Override
   public void dispose() {
     disposing = true;
-    sinks.asMap().values().forEach(sink -> sink.complete());
-    sinksNestedTx.asMap().values().forEach(sink -> sink.complete());
+    sinks.asMap().values().forEach(sinkList -> sinkList.forEach(FluxSinkWrapper::complete));
+    legacySinks.asMap().values().forEach(FluxSink::complete);
+    sinksNestedTx.asMap().values().forEach(sinkList -> sinkList.forEach(FluxSinkWrapper::complete));
+    legacySinksNestedTx.asMap().values().forEach(FluxSink::complete);
```

Comment:
I think we are missing the invalidateAll in the legacy ones down below on lines 200-201 and 211-212 (it doesn't let me put the commend directly there)

Reply:
Yes. Sorry. Missed that.